src/openms_gui/source/VISUAL/Spectrum1DCanvas.cpp

```diff
@@ -1130,7 +1130,7 @@ namespace OpenMS
     {
       (*getCurrentLayer_().getPeakDataMuteable())[i].sortByPosition();
     }
-    getCurrentLayer_().getCurrentSpectrum().sortByPosition();
+    getCurrentLayer_().sortCurrentSpectrumByPosition();
```

Comment:
not sure how necessary this is here, but this does the same thing as the code did before

Reply:
If we use our default mzML reader the spectra should be sorted on load (I think I implemented this quite a while ago). Question is, if this is also the case for the OnDisc data. If so we would not need mutable access here I guess

Reply:
we currently do not, we could add it here: https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/HANDLERS/MzMLSpectrumDecoder.cpp#L310 and here https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/FORMAT/HANDLERS/MzMLSpectrumDecoder.cpp#L343

Reply:
what do you think? should we do so?
Could otherwise lead to different behavior if in memory and on disc is switched

Reply:
> should we do so?

yes, we should - there is no reason not to